COORDINATE_SYSTEM2/COORDINATE_SYSTEM2/CoordinateSystem.cs

```diff
@@ -6,21 +6,89 @@
 
 namespace COORDINATE_SYSTEM2
 {
+    /// <summary>
+    /// Клас Координатна система.
+    /// Містить Ліст всіх створених точок та Ліст всіх знайдених дистанцій.
+    /// Містить Add(), який створює нову точку, добавляє її в Ліст, шукає відстані 
+    /// між існуючими точками до нової точки і добавляє їх в ліст.
+    /// Містить DisplayAllPoints(), який друкує всі точки з ліста.
+    /// Містить MaxDistance() для пошуку максимальної відстані між точками.
+    /// Містить PrintAllDistances(), який друкує всі наявні відстані з Ліста.
+    /// </summary>
     internal class CoordinateSystem
     {
         List<Point> listOfPoints = new List<Point>();
 
+        List<Rectangle> listOfRectangles = new List<Rectangle>();
+
+        List<Triangle> listOfTriangles = new List<Triangle>();
+
+        List<IFigure> listOfFigures = new List<IFigure>();
+
         List<PointsDistance> pointsDistances = new List<PointsDistance>();
 
-        public void Add(Point point)
+        //TODO мені потрібно зробити КЛАС!!! "ФІГУРА"
+        //- який буде базовим для Точки, Прямокутника та Трикутника.
+        //І в методі Ед приймати їх через цей базовий клас.
+        public void Add(IFigure figure) 
+        {
+            if(figure is Point)
+            {
+                foreach (Point point1 in listOfPoints)
+                {
+                    PointsDistance distance = new PointsDistance((Point)figure, point1);
+                    pointsDistances.Add(distance);
+                }
+
+                listOfPoints.Add((Point)figure);
+            }
```

Comment:
You can do next magic

```suggestion
            if(figure is Point pointFigure)
            {
                foreach (Point point1 in listOfPoints)
                {
                    PointsDistance distance = new PointsDistance(pointFigure, point1);
                    pointsDistances.Add(distance);
                }

                listOfPoints.Add(pointFigure);
            }
```

Reply:
this code has to be rewritten anyway)) the shouldn't be any checks on type